docs/source/grid_generation/explanation/infdb_buildings_processor.rst

```diff
@@ -0,0 +1,184 @@
+
+Data Sources
+------------
+
+The processor uses three InfDB data sources:
+
+- **3DCityDBv5**: Building data
+- **Census**: Statistical data in 100m x 100m grids
+- **Basemap**: Mainly used for processing streets
+
+3DCityDBv5 resides in the ``citydb`` schema. Census and Basemap are located in the ``opendata`` schema.
+
+A visual overview of the column sources for ``pylovo_input.buildings`` is shown below:
+
+.. image:: ../../images/infdb/buildings_sources.jpg
+    :width: 100%
+    :alt: Data sources
+
+- ``id``: From ``citydb.feature.id``
+- ``objectid``: From ``citydb.feature.objectid``
+- ``geom``: From ``citydb.geometry_data.geometry``
+- ``building_use_id``, ``building_use``, ``floor_area``, ``height``: From ``citydb.property.val_*``
+- ``floor_number``: From ``citydb.property.val_*`` or estimated from ``height`` if missing
+- ``construction_year``: From ``opendata.cns22_100m_baujahr_jz``
+- ``building_type``: Derived from ``opendata.cns22_100m_wohnung_gbtyp_groesse`` plus ``height`` and ``floor_area``
+- ``occupants_per_building``: From ``opendata.cns22_100m_bevoelkerungszahl``
+- ``households_per_building``: From ``opendata.cns22_100m_durchschn_haushaltsgroesse``
+- ``postcode``: From ``opendata.plz_plz-5stellig.plz``
+
+Column Roles
+------------
+
+- **id**: Building ID
+- **objectid**: Alternate unique ID (from citydb)
+- **geom**: 2D geometry in EPSG:3035
+- **building_use_id**: Internal ID for building use
+- **building_use**: One of ``Residential``, ``Industrial``, ``Commercial``, ``Public``
+- **floor_area**: Ground floor area
+- **height**: Building height
+- **floor_number**: Number of floors
+- **construction_year**: Ranges like ``'-1919'``, ``'1919-1948'`` up to ``'2020-'``
+- **building_type**: ``AB``, ``MFH``, ``TH``, or ``SFH``
+- **occupants_per_building**: Estimated number of residents
+- **households_per_building**: Estimated number of households
+- **postcode**: Postcode based on building centroid
+
+Data Filling Process
+--------------------
+
+An overview of the processing steps can be seen in the following visualization:
+
+.. image:: ../../images/infdb/buildings_steps.jpg
+    :width: 100%
+    :alt: Processing steps
+
+#. Create the ``buildings`` table.
+#. Fill ``id``, ``objectid``, ``building_use``, ``building_use_id``, ``height``, ``floor_area``, and ``geom`` using CityDB.
+#. Remove buildings that are too small to reasonably use electricity.
+#. Fill ``floor_number`` directly or estimate using building height and median floor heights.
+#. Use CityDB, Census, and nearest-grid data to fill ``occupants``, ``households``, and ``construction_year``.
+#. Classify ``building_type`` using ``floor_number``, ``floor_area``, and neighbor analysis.
+#. Fix inconsistencies, e.g., SFHs must have only one household.
+#. Rebalance ``building_type`` distribution to match census data.
+#. Assign postcodes using PLZ geometry.
+#. Assign ``address_street_id`` using addresses from buildings and ways (nullable).
+
+Assumptions
```

Comment:
I think in general it would be easier to understand if you connect the steps 1-10 to the assumptions relevant for these steps. Or you link them better. Otherwise it is kind of confusing.